File: doc-examples/jdbc-example-records-java/src/main/java/example/Student.java
package example;

import edu.umd.cs.findbugs.annotations.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;

import java.util.Collections;
import java.util.List;
import java.util.Set;

@MappedEntity
public record Student(@Id @GeneratedValue @Nullable
                      Long id,
                      String name,
                      @Relation(value = Relation.Kind.MANY_TO_MANY, cascade = Relation.Cascade.PERSIST)
                      List<Course> courses,
                      @Relation(value = Relation.Kind.ONE_TO_MANY, mappedBy = "student")
                      Set<CourseRating> ratings) {

    public Student(String name, List<Course> courses) {
        this(null, name, courses, Collections.emptySet());
    }

}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/Student.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation

@MappedEntity
data class Student(
        @field:Id @GeneratedValue
        val id: Long?,
        val name: String,
        @Relation(value = Relation.Kind.MANY_TO_MANY, cascade = [Relation.Cascade.PERSIST])
        val courses: List<Course>,
        @Relation(value = Relation.Kind.ONE_TO_MANY, mappedBy = "student")
        val ratings: List<CourseRating>
) {
    constructor(name: String, items: List<Course>) : this(null, name, items, emptyList())
}
